File: workspace/2023-08-07/numerical-sequence.prolog/fizz-buzz.pl

fizz_buzz(N, Fn) :-
    N mod 15 =:= 0 ->
        Fn = "FizzBuzz";
    N mod 5 =:= 0 ->
        Fn = "Buzz";
    N mod 3 =:= 0 ->
        Fn = "Fizz";
    otherwise -> 
        number_string(N, Fn).

File: workspace/2023/2023-01-26/fizz-buzz.haskell/fizz-buzz.hs

main :: IO ()
main = print $ fizzBuzz 100

fizzBuzz :: Int -> [String]
fizzBuzz n = map fizzBuzzValue [1..n]

fizzBuzzValue :: Int -> String
fizzBuzzValue n | n `mod` 15 == 0 = "FizzBuzz"
                | n `mod`  5 == 0 = "Buzz"
                | n `mod`  3 == 0 = "Fizz"
                | otherwise       = show n

